torchvision/prototype/transforms/_misc.py

```diff
@@ -79,3 +81,28 @@ def __init__(self, dtype: torch.dtype, *types: Type) -> None:
 
     def extra_repr(self) -> str:
         return ", ".join([f"dtype={self.dtype}", f"types={[type.__name__ for type in self.types]}"])
+
+
+class CleanupBoxes(Transform):
+    def _get_params(self, sample: Any) -> Dict[str, Any]:
+        bounding_boxes = query_bounding_box(sample)
+        bounding_boxes_clamped = F.clamp_bounding_box(
+            bounding_boxes, format=bounding_boxes.format, image_size=bounding_boxes.image_size
+        )
+        bounding_boxes_xywh = F.convert_bounding_box_format(
+            bounding_boxes_clamped, old_format=bounding_boxes.format, new_format=features.BoundingBoxFormat.XYWH
+        )
+        is_valid = torch.all(bounding_boxes_xywh[..., 2:] > 0, dim=-1)
+
+        return dict(is_valid=is_valid)
```

Comment:
I have the feeling that this transform might not fit nicely in the API. You effectively want to clamp the bboxes first and modify their values and then based on this outcome decide that is invalid sync the rest of the items in the output.

Reply:
Yes, I would also implement only forward for this transform